### PR TITLE
Using range method to check status code

### PR DESCRIPTION
**Pull request by nikolasvargas**
More readable

### CODE REVIEW

**Comment by sigmavirus24** [comment]
But it's not

**Comment by nikolasvargas** [comment]
ok

File: requests/models.py [hunk 1]
```diff
@@ -918,7 +918,7 @@ def links(self):
     def raise_for_status(self):
         """Raises :class:`HTTPError`, if one occurred."""
 
-        http_error_msg = ''
+        http_error_msg, reason = '', ''
```
Comment (AaronRobson) [inline requests/models.py:921]: This change appears to be unrelated to the range change.
Was it suggested by a linting tool?

**Comment by nateprewitt** [comment]
Hi @nikolasvargas,

Thanks for the contribution! I don't know if I agree this is more readable though. It has the additional negative that we generate a bunch of additional numbers in the range we have to iterate but will never possibly use. The current implementation is pretty direct and common usage for comparison operators in Python.

I think we'll leave this as is for now.
